lib/iris/__init__.py

```diff
@@ -197,7 +197,7 @@ def __repr__(self):
         return msg.format(self.cell_datetime_objects, self.netcdf_promote,
                           self.netcdf_no_unlimited, self.clip_latitudes)
 
-    deprecated_options = {}
+    deprecated_options = {'cell_datetime_objects'}
```

Comment:
Have you confirmed this works as required? There's a very definite statement about how this must behave in #2831, which must be observed.